### PR TITLE
[quant][pt2e] Add reference representation for quantized conv2d

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #105784
* #105783

Summary:
Implementing reference representation for quantized ops we decided in https://docs.google.com/document/d/17h-OEtD4o_hoVuPqUFsdm5uo7psiNMY8ThN03F9ZZwg/edit#heading=h.ov8z39149wy8

Test Plan:
python test/test_quantization.py TestQuantizePT2E.test_representation_quantize_dequantize_per_channel

Although right now it is not really testing things since there is some problem with dynamo export

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

File: torch/ao/quantization/pt2e/representation/rewrite.py [hunk 1]
```diff
@@ -14,6 +14,64 @@
...
+    weight_i16 = weight_i8.to(torch.int16)
+    # always set bias to None so that the same representation can work for the case no matter if bias_scale == x_scale * weight_scale or not
+    acc_i32 = out_dtype(torch.ops.aten.convolution.default, torch.int32, x_i16 - x_zero_point, weight_i16 - weight_zero_point, None, stride, padding, dilation, transposed, output_padding, groups)
+    # TODO: change to mul.Scalar when we make x_scale/weight_scale etc. Scalar values
+    acc_i32 = out_dtype(torch.ops.aten.mul.Tensor, torch.int32, acc_i32, x_scale * weight_scale / out_scale)
+    # TODO: change to mul.Scalar
+    # Note: we are quantizing bias with these scales without signal from user, but it might be OK
+    bias_scale = x_scale * weight_scale
+    bias_i32 = out_dtype(torch.ops.aten.mul.Tensor, torch.int32, bias_fp32, bias_scale / out_scale)
+    out_i8 = torch.ops.aten.clamp(acc_i32, out_quant_min, out_quant_max).to(torch.int8)
```
Comment (kimishpatel) [inline torch/ao/quantization/pt2e/representation/rewrite.py:71]: You havent added bias to acc_i32

Comment (kimishpatel) [reply torch/ao/quantization/pt2e/representation/rewrite.py:71]: Also, requantization happens after adding bias. and you need to add out_zero_point too.

**Review by kimishpatel — CHANGES_REQUESTED** [review]

**Review by kimishpatel — APPROVED** [review]
unblock:

please add comments, that we landing this to unblock but this needs to be vetted since these are not functionally tested and some comments are not addressed

**Comment by jerryzh168** [comment]
> unblock:
> 
> please add comments, that we landing this to unblock but this needs to be vetted since these are not functionally tested and some comments are not addressed

I have tested these locally and fixed the bias stuff as well, will probably enable the tests in CI soon since I didn't find it is timing out in sandcastle recently. what is the not addressed comment are you referring to btw?

**Comment by jerryzh168** [comment]
TODO: we need to support autograd for out_dtype op so that user don't need to use torch.no_grad to run the quantized model.

**Pull request closed without merging**